pilot/pkg/networking/core/tracing.go

```diff
@@ -462,6 +495,116 @@ func otelLightStepConfig(clusterName, hostname, accessToken string) (*anypb.Any,
 	return anypb.New(dc)
 }
 
+func configureDynatraceSampler(hostname, cluster string,
+	otelProvider *meshconfig.MeshConfig_ExtensionProvider_OpenTelemetryTracingProvider,
+	sampler *meshconfig.MeshConfig_ExtensionProvider_OpenTelemetryTracingProvider_DynatraceSampler_,
+	pushCtx *model.PushContext) (*core.TypedExtensionConfig, error) {
+
+	dsc := &otelsamplers.DynatraceSamplerConfig{
+		Tenant:             sampler.DynatraceSampler.GetTenant(),
+		ClusterId:          sampler.DynatraceSampler.GetClusterId(),
+		RootSpansPerMinute: sampler.DynatraceSampler.GetRootSpansPerMinute(),
+	}
+
+	if sampler.DynatraceSampler.HttpService == nil {
+		// The Dynatrace sampler can re-use the same Dynatrace service/cluster/auth header
+		// as configured for the HTTP Exporter. In this case users
+		// can achieve a much smaller/simpler config in Istio.
+		var header *meshconfig.MeshConfig_ExtensionProvider_HttpHeader
+
+		// Re-use the Dynatrace API Host + Token from the HTTP exporter
+		httpService := otelProvider.GetHttp()
+		if httpService == nil {
+			return nil, fmt.Errorf("dynatrace sampler could not get http settings. considering using dynatrace_api field")
+		}
+
+		uri, err := url.JoinPath(hostname, "api/v2/samplingConfiguration")
+		if err != nil {
+			return nil, fmt.Errorf("could not parse dynatrace adaptative sampling endpoint %v", err)
+		}
+
+		// Extracts the Authorization header from the HTTP exporter headers
+		for _, h := range httpService.GetHeaders() {
+			if h.Name != "Authorization" {
+				continue
+			}
+			hv := h.GetValue()
+			if !strings.Contains(strings.ToLower(hv), "api-token ") {
+				return nil, fmt.Errorf("could not parse dynatrace token from http header")
+			}
+			if len(strings.Fields(hv)) != 2 {
```

Comment:
Are we sure that our tokens will always have the format `api-token blabla`?

Reply:
I'm not aware of any changes, and I would be surprised if we decide to change that now, since it's so widespread.

Reply:
I don't think that we will change the format now or soon. But if we ever do, I am quite sure that we will forget to update this check ;-). But no strong opinion on that.